src/plugins/intel_cpu/src/utils/debug_capabilities.h

```diff
@@ -15,12 +15,71 @@
 
 #include <onednn/dnnl.h>
 #include <dnnl_debug.h>
+#include <cpu/x64/jit_generator.hpp>
 
 #include "openvino/core/model.hpp"
 
 namespace ov {
 namespace intel_cpu {
 
+class RegPrintManager;
+
+#define PRINT_REGISTER_SELECTOR(_1, _2, name, ...) name
+
+#define PRINT_REGISTER_WITH_VALUE(REG_VAL)             \
+{                                                      \
+    RegPrintManager manager(*this, REG_VAL, #REG_VAL); \
+    manager.PrintRegister(); \
+}
+
+#define PRINT_REGISTER_WITH_VALUE_PRECISION(REG_VAL, REG_PRECISION)   \
+{                                                                     \
+    RegPrintManager manager(*this, REG_VAL, #REG_VAL, REG_PRECISION); \
+    manager.PrintRegister();                                          \
+}
+
+#define PRINT_REGISTER(...)                                      \
+    PRINT_REGISTER_SELECTOR(__VA_ARGS__,                         \
+                            PRINT_REGISTER_WITH_VALUE_PRECISION, \
+                            PRINT_REGISTER_WITH_VALUE)(__VA_ARGS__)
+
+class RegPrintManager {
```

Comment:
Some general thoughts. 
1. From architecture perspective I would prefer to have this functionality incapsulated into base class for all JIT functionality. We have a plan to create base class for all JIT kernels and JIT emitters inside the plugin and put this auxiliary staff inside. Since now we don't have such hierarhy lets leave as a separate class.
2. Given 1st statement it is better to avoid creation of objects of this class. My proposal it to keep all print methods static and hide costructors, while fields should be const expression. This is very simple printing functionality, so it shouldn't have any state.
3. Maybe we can call it RegPrinter and have print public methods like:
```
template< typename PRC_T, typename REG_T>
static void print(jit_generator &h, REG_T reg)
```
Then usage might look like:
```
RegPrinter::print<float>(*this, vmm_src);
```
Precision type is deduced from template param (PRC_T). Such approach will allow you to specialize template for only required precisions and avoid functions like CheckPrecision().
Register type is deduced from function argument (REG_T). It wil allow you to avoid checks like 
```
reg_val.isREG(64) || reg_val.isREG(32) || reg_val.isREG(16) || reg_val.isREG(8)
```
and just use template specialization mechanism.

4. Given 1st statement I would prefer to avoid MACRO usage. It was introduced by two reasons as I can see: object creation and register name printing. First is solved by static print method, second can be solved by overloading print method with additional register name attribute:
```
template< typename PRC_T, typename REG_T>
static void print(jit_generator &h, REG_T reg, std::string name)
```
By default if name is not passed you can just print number of register and its type getting it from Xbyak. So if vmm_src is actually YMM0 you can print exactly YMM0 before register content.

What do you think? If you have any concerns feel free to discuss.

Reply:
@dmitry-gorokhov Hi Dmitry, I like your proposal! And I want to make sure that I think it the same way as you do for your first point. By saying "From architecture perspective I would prefer to have this functionality incapsulated into base class for all JIT functionality", do you mean, for example, to revise
`struct jit_uni_topk_kernel_f32 : public jit_uni_topk_kernel, public jit_generator` to
`struct jit_uni_topk_kernel_f32 : public jit_uni_topk_kernel, public jit_generator,  public RegPrinter`
and apply such revision to all the other jit kernals of CPU plugin nodes?

Reply:
@xuchen-intel My first statement was about view on future achitecture. This is not needed for current PR.

In short I expect to have jit_base class that is derived from jit_generator. All Jit kernels and emitters will be derived from jit_base. This jit_base will have print capability implemented as its protected method.

Reply:
Good suggestions! Applied. Thanks Dmitry!